Feat/observation unit export

jnizet opened the pull request:
On top of #270

jnizet committed:
feat: implement MIAPPE export of germplasms

jnizet committed:
feat: export observation units

jnizet committed:
feat: implement repository methods to list selectable search criteria

jnizet committed:
feat: implement UI for exports

But the requests for large Excel files (or even for CSV when there are a lot of data to export) fail with a timeout.
We could increase the timeout time (using spring.mvc.async.request-timeout), but a proper solution would probably be to

- create a job that writes the export to a temporary file in a separate thread, identified by some ID.
- respond immediately with this export ID
- have the frontend navigate to a new page (or stay in the same one, but it's probably simpler and safer to navigate to a new page) with this ID as parameter
- have this new page poll the job until it's done, and finally download the generated file.

cexbrayat commented on backend/src/main/java/fr/inrae/urgi/faidare/web/observationunit/ObservationUnitExportService.java:
```diff
@@ -40,17 +42,11 @@
         List<Column> columns = createColumns(rows);
 
-        try (SXSSFWorkbook workbook = new SXSSFWorkbook(100)) {
+        try (SXSSFWorkbook workbook = new SXSSFWorkbook(1)) {
```
Not sure what this changes, just flagging it in case this change was unintended

cexbrayat commented on backend/src/main/java/fr/inrae/urgi/faidare/web/trial/TrialController.java:
```diff
@@ -83,13 +92,22 @@ public ModelAndView get(@PathVariable("trialId") String trialId, HttpServletRequ
         // Clarification: the spec was not fully explicit — the goal is to show the 4 fields from the screenshot (name, source, type, description) plus the location.
         // Agreed: no need to use a BrAPI endpoint — using the DAO directly is fine here.
```
unrelated, but maybe the above comments should be removed ?

cexbrayat commented on web/src/trial/trial-export.ts:
```diff
@@ -0,0 +1,79 @@
+            spinner.classList.add('d-none');
+            progress.classList.add('d-none');
+        }
+    });
```
This file could use some comments, and is not formatted as we usually do.
Maybe we could add the lint/formatter, and maybe we could add an e2e test with playwright? WDYT?

jnizet replied:
Yes, we should. I will.

jnizet committed:
chore: formatting

jnizet committed:
feat: make export asynchronous

jnizet committed:
feat: format backend templates

jnizet committed:
test: add e2e tests

jnizet committed:
feat: create rows from exported observation units correctly

also

- fix download button style
- include study location in observation unit export criteria

jnizet committed:
feat: delete leftover export files at startup

jnizet committed:
refactor: rename observation unit export to observation export

jnizet committed:
feat: export observations as JSON

jnizet committed:
fix: split searches with large set of IDs in batches

jnizet committed:
fix: fix bugs in DAO implementations of sitemap queries and add tests
